distributed/scheduler.py

```diff
@@ -2535,9 +2537,26 @@ def worker_send(self, worker, msg):
             else:
                 who_has[key] = []
 
-        data, missing_keys, missing_workers = await gather_from_workers(
-            who_has, rpc=self.rpc, close=False, serializers=serializers
-        )
+        missing_keys = who_has
+        missing_workers = set(self.workers)
+        iteration = 0
+        while missing_workers:
+            missing_workers_begin = missing_workers.copy()
+            missed_data, missing_keys, missing_workers = await gather_from_workers(
```

Comment:
Why the name `missed_data`?  It seems like this is the data that we aren't missing.

Reply:
`naming is the hardest problem in computer science`

I will have another look at it :)